miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java

```diff
@@ -279,6 +284,42 @@ public PoolPickerResponse getPickersBySearch(@RequestParam("platform") String pl
         PaginationFilter.platformType(PlatformType.valueOf(platform)));
   }
 
+  @RequestMapping(value = "avginsertsizes", method = RequestMethod.POST, produces = "application/json")
+  @ResponseBody
+  public JSONObject calculateAverageInsertSizes(@RequestBody JSONObject json) throws IOException {
+    String idString = json.getString("ids");
+    String[] split = idString.split(",");
+    List<Long> ids = new ArrayList<>();
+    for (int i = 0; i < split.length; i++) {
+      ids.add(Long.parseLong(split[i]));
+    }
+    Collection<Pool> pools = new ArrayList<>();
+    pools = poolService.listByIdList(ids);
+    List<String> avgs = new ArrayList<>();
+    for (Pool p : pools) {
+      long sum = 0;
+      Set<PoolableElementView> pevs = p.getPoolableElementViews();
+      if (pevs.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)